### ISSUE TITLE
0.4.7 is backwards-incompatible

### ISSUE THREAD

**Issue by darkwater** [ynqa/promkit#34]
I tried to `cargo install jnv`, but it failed with this error:

```
error[E0053]: method `finalize` has an incompatible type for trait
   --> /home/dark/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jnv-0.4.1/src/jnv.rs:254:17
    |
254 |     fn finalize(&self) -> anyhow::Result<Self::Return> {
    |                 ^^^^^ types differ in mutability
    |
    = note: expected signature `fn(&mut Jnv) -> Result<_, _>`
               found signature `fn(&Jnv) -> Result<_, _>`
help: change the self-receiver type to match the trait
    |
254 |     fn finalize(&mut self) -> anyhow::Result<Self::Return> {
    |                 ~~~~~~~~~

For more information about this error, try `rustc --explain E0053`.
error: could not compile `jnv` (bin "jnv") due to 1 previous error
```

Digging into the code, this seems to be because `promkit` changed the API in [000ea72e: change finalize to `mut self`](https://github.com/ynqa/promkit/commit/000ea72ecc18089432752e112b71ffc1155d6dc0)

This is a backwards-incompatible change, so I believe 0.4.7 should be yanked and published as 0.5.0 instead.

**Comment by ynqa** [comment]
@darkwater Thank you for the report, I understand that. The approach seems fine as well.

**Comment by ynqa** [comment]
Yanked [v0.4.7](https://crates.io/crates/promkit/0.4.7) and released [v0.5.0](https://github.com/ynqa/promkit/releases/tag/v0.5.0).
